src/test/java/com/puppycrawl/tools/checkstyle/utils/TokenUtilTest.java

```diff
@@ -321,6 +330,31 @@ public void testForEachChild() {
             .isEqualTo(secondSibling);
     }
 
+    @Test
+    public void testForEachChildMultipleTypes() throws Exception {
+        final DetailAST astForTest = getNodeFromFile(TokenTypes.OBJBLOCK,
+                "InputTokenUtilTestForEachChildMultipleTypes.java");
+        final List<String> children = new ArrayList<>(3);
+        final List<Integer> tokensToMatch = List.of(TokenTypes.CLASS_DEF, TokenTypes.VARIABLE_DEF);
+
+        TokenUtil.forEachChild(astForTest, tokensToMatch, child -> {
```

Comment:
I might was no clear. I know it is very unusual and sattle.
We allow tests only by Inputs with embedded config.
All other forms of testing are well proven "way to hell" in our repo .

Please no updates in TokenUtilTest.

Let's come back to "map.or" solution , sorry that we confused you and lost valuable time.

Let me know if you interested in more details why we follow unusual philosophy

Reply:
> Inputs with embedded config.

This is not possible here since it is not a check/filter itself.

---
> Please no updates in TokenUtilTest
> Let's come back to "map.or" solution 

Reverted back. Done.

---
It's alright. I would not call it wasted time. Dug around the codebase and learned a few things about the rules of this repository and how pull requests should be.


> All other forms of testing are well proven "way to hell" in our repo .
>  Let me know if you interested in more details why we follow unusual philosophy

Yes, I'd like to know why. In my eyes, such a test does not seem harder to maintain than any other check's tests and input files.

Reply:
We try our best to use tests only like `verifyWithInlineConfigParser(getNonCompilablePath(filename)`.

It makes us confident that we always use compilable input files. And all code in Check can handle any possible-and-compilable combinations of java code and all code in Check are there for a reason (pitest) no extra code and no dead code. Removal/mutation of any part of it immediately detected by Junit tests.

As you might notice in above statement we do not mention about util methods at all , as potentially until method can be copied to class and be part of it. It is our details of implementation that we use util methods, we do not want to let anyone know it, and it subject to change at any time.

If we allow testing of utility methods separately, all logic of all Checks can be generalized and moved to utilities so whole test pressure will also move to utilities. And we will struggle with auto measure of quality of Check. 
In result maintainers will win in short term, user will loose in long term, usage will drop,  maintainers will not found way to invite new contributions, project is dead eventually.

Right now as you might noticed, new contributor does not need to know all ast and all utilities and all rules. All you need is to put a java file to test and place debug in Check visitToken.

Reply:
We did not born smart, it took us 20 years to do all types of testing to end up where bee are. So are of code is still old style

Reply:
I get it now - all in the name of maintainability and keeping the project going and easy to contribute to. Which I have noticed btw. 

What does that mean for the util tests though? Do you want them to move over to such testing `verifyWithInlineConfigParser`? Is it even possible? Or do you expect tests from checks that use util methods to cover them?